File: pulse-kotlin/src/commonTest/kotlin/kotlin/com/fluxchess/pulse/kotlin/CastlingTest.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package kotlin.com.fluxchess.pulse.kotlin

import com.fluxchess.pulse.kotlin.castlings
import kotlin.test.Test
import kotlin.test.assertEquals

class CastlingTest {
	@Test
	fun `Castlings array should be valid`() {
		for (castling in castlings.indices) {
			assertEquals(castlings[castling], castling)
		}
	}
}

File: pulse-go/internal/pulse/castling_test.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package pulse

import (
	"testing"
)

func TestCastling(t *testing.T) {
	t.Run("Castlings array should be valid", func(t *testing.T) {
		for castling := range Castlings {
			if Castlings[castling] != castling {
				t.Errorf("wanted castlings at index %d to be %v, got %v", castling, castling, Castlings[castling])
			}
		}
	})
}
